### PR TITLE
Close Space event payload, properly remove confirmation modal element on close, and ensure duplicate events are not fired when opening multiple Spaces

### PR DESCRIPTION
**Pull request by NateFerrero**
## Please explain how to summarize this PR for the Changelog:

- Ensures that the event for closing the space is only fired once
- Ensures that when closing a space, opening a new space, and closing the new space, that the close event listener from the first space is unregistered, to avoid duplicate events being handled
- Ensures that the close confirmation modal element is cleanly removed after the space is closed

## Tell code reviewer how and what to test:

Test both preloaded and on-demand spaces in `apps/vanilla`, you should see the following close event payload in the browser console after clicking "Continue" after submitting your data:

Screenshot from 2024-06-25 22-09-40

### CODE REVIEW

**Comment by NateFerrero** [comment]
I'm not 100% sure this resolves https://github.com/FlatFilers/support-triage/issues/1393

Note that the only event that can close the Space is clicking the Continue button here:
image

Per the code:
```TypeScript
flatfileEvent.topic === 'job:outcome-acknowledged' &&
flatfileEvent.payload.status === 'complete' &&
flatfileEvent.payload.operation === closeSpace?.operation
```
And `payload` looks like this:
```JavaScript
{
  "job": "workbook:submitActionFg",
  "info": "Getting started.",
  "domain": "workbook",
  "isPart": false,
  "status": "complete",
  "operation": "submitActionFg"
},
```

So it seems in order to take advantage of this, `closeSpace` in the Flatfile options must be set up as such:

```JavaScript
closeSpace: {
  operation: 'submitActionFg',
  onClose: (event) => {
    console.log(
      `Close space event payload: ${JSON.stringify(event, null, 2)}`
    )
  },
},
```

**Comment by NateFerrero** [comment]
Should fix https://github.com/FlatFilers/support-triage/issues/1382

**Comment by NateFerrero** [comment]
Should fix https://github.com/FlatFilers/support-triage/issues/1408

File: packages/javascript/FlatfileJavascript.ts [hunk 1]
```diff
@@ -179,20 +176,34 @@ function initializeIFrameConfirmationModal(
   confirmModal.style.display = 'none'
   document.body.appendChild(confirmModal)
 
-  window.addEventListener(
-    'message',
-    (event) => {
+  function closeFlatfileSpace(event: any) {
```
Comment (bangarang) [inline packages/javascript/FlatfileJavascript.ts:179]: We should likely use a similar re-usable pattern like we do in `handlePostMessage from @flatfile/embedded-utils`. Can be a fast follow!

**Comment by bangarang** [comment]
This looks great! I added a changeset file with the `npm run changeset` command here - for future reference @NateFerrero

**Review by bangarang — APPROVED** [review]

File: packages/javascript/FlatfileJavascript.ts (hunk 1 above)
Comment (NateFerrero) [reply packages/javascript/FlatfileJavascript.ts:179]: Done in #135 @bangarang !

**Pull request merged**